### DISCUSSION TITLE
Can't use .includes() inside _middleware

### DISCUSSION THREAD

**Discussion by Rees1993** [vercel/next.js#33152] [Help]
When creating _middleware and deploying on Vercel, it appears that you can't use `.includes()` to check if a string exists in an array. I was using the code below code to check against multiple pages. This seems to work on local dev.

```
import { getToken } from "next-auth/jwt";
import { NextResponse } from "next/server";

/** @param {import("next/server").NextRequest} req */
export async function middleware(req) {
  const protectedPages = ["/route-1", "/route-2", "/route-3", "/route-4"];

 if (protectedPages.includes(req.nextUrl.pathname)) {
    const session = await getToken({
      req,
      secret: process.env.SECRET,
      secureCookie: process.env.NEXTAUTH_URL?.startsWith("https://") ?? !!process.env.VERCEL_URL,
    });

    if (!session) return NextResponse.redirect("/login");
  }
}

```

If I replace the `if (protectedPages.includes(req.nextUrl.pathname))` with the following, which checks each one individually, then it works.

```
  if (
    req.nextUrl.pathname === "/route-1" ||
    req.nextUrl.pathname === "/route-2" ||
    req.nextUrl.pathname === "/route-3" ||
    req.nextUrl.pathname === "/route-4"
  )
```

Is this a limitation of _middleware as I am aware that you can't do some things inside of them? 

URL: https://next-auth-example-ar.vercel.app/
Repo: https://github.com/Rees1993/nextjs-middleware-includes

**Comment by balazsorban44** [comment]
What do you mean by not working? Any error thrown or a seemingly logical error?

You use `href` in the `includes` method but `pathname` in the other. I belive your if statement simply never matches any of the options

**Reply by Rees1993** [reply]
Sorry, should have been more helpful than just not working. 

I can't see any errors being thrown inside of the function log and it is working correctly when you are working locally. But on production, when hosted on Vercel, you get to get to middleware protected pages without having the authentication needed.

Also I have edited the example as they should both be using pathname as they are in my example.
